src/server/pfs/db/driver.go

```diff
@@ -1322,7 +1322,12 @@ func (r *fileReader) Read(data []byte) (int, error) {
 			break
 		}
 		client := client.APIClient{BlockAPIClient: r.blockClient}
-		r.reader, err = client.GetBlock(blockRef.Hash, uint64(r.offset), uint64(r.size))
+		sizeLeft := r.size
+		// e.g. sometimes a reader is constructed of size 0
+		if r.size > r.sizeRead {
```

Comment:
This should always be true unless `r.size` is `0` right?
Assuming that's true I think it'd be clearer to just do: `if sizeLeft != 0`